Submitted By:     VMiklos <vmiklos@frugalware.org>
Date:             2005.04.10
Initial Packages: 2.9.5
Upstream Status:  Not submitted - Frugalware specific
Description:      This patch adds a new feature to makepkg.
                  If chkdep available, search for package dependencies
                  automatically if you use the -a switch.

--- pacman-2.9.5.orig/scripts/makepkg	2005-04-07 13:02:01.000000000 +0200
+++ pacman-2.9.5/scripts/makepkg	2005-04-07 13:09:56.000000000 +0200
@@ -178,6 +178,7 @@
 	echo "makepkg version $myver"
 	echo "usage: $0 [options]"
 	echo "options:"
+	echo "  -a, --searchdeps Search for package dependencies automatically"
 	echo "  -b, --builddeps  Build missing dependencies from source"
 	echo "  -c, --clean      Clean up work files after build"
 	echo "  -C, --cleancache Clean up source files from the cache"
@@ -221,6 +222,7 @@
 
 while [ "$#" -ne "0" ]; do
 	case $1 in
+		--searchdeps) SEARCHDEPS=1 ;;
 		--clean)      CLEANUP=1 ;;
 		--cleancache) CLEANCACHE=1 ;;
 		--syncdeps)   DEP_BIN=1 ;;
@@ -243,8 +245,9 @@
 			exit 1
 			;;
 		-*)
-			while getopts "bBcCdefghij:Lmnop:rsSw:-" opt; do
+			while getopts "abBcCdefghij:Lmnop:rsSw:-" opt; do
 				case $opt in
+					a) SEARCHDEPS=1 ;;
 					b) DEP_SRC=1 ;;
 					B) NOCCACHE=1 ;;
 					c) CLEANUP=1 ;;
@@ -735,6 +738,19 @@
 $cmd | sort >../filelist
 
 cd $startdir
+if [ "$SEARCHDEPS" == "1" ]; then
+	if [ -x /usr/bin/chkdep ]; then
+		msg "Searching for depends..."
+		warning "Do not use this information without thinking!"
+		chkdep -d pkg/ -n $pkgname -i
+	else
+		warning "Chkdep is not installed and is required to search depends."
+		plain "Install the pacman-tools package, wich contains chkdep."
+		plain ""
+		sleep 1
+	fi
+fi
+
 if [ "$CLEANUP" = "1" ]; then
 	msg "Cleaning up..."
 	rm -rf src pkg filelist
